docs/components/file-upload.md

```diff
@@ -0,0 +1,213 @@
+    </KFileUpload>
+  </template>
+</KCard>
+
+```html
+<KFileUpload appearance="image" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" placeholder="You can change the text here!" icon="kong" />
+```
+
+### removable
+
+A `cancel` button can be displayed for `file` type upload, by default this is set to `false`. This button is only visible once a file has been added.
+
+
+### accept
+
+Please refer to [mdn web docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept) that defines the file types the file input should accept. 
+
+### buttonAppearance
+
+Use this prop to customize the trigger **KButton** [appearance](/components/button.html#appearance). Default set to `primary`.
+
+<KCard>
+  <template v-slot:body>
+    <KFileUpload label="Upload File" buttonAppearance="creation" :label-attributes="{ help: 'I use the KLabel `help` prop' }" help="Additional text." icon="close" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" appearance="file" removable />
+  </template>
+</KCard>
+
+```html
+<KFileUpload label="Upload File" buttonAppearance="creation" :label-attributes="{ help: 'I use the KLabel `help` prop' }" help="Additional text." icon="close" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" appearance="file" removable />
+```
+
+### buttonText
+
+This is the text that will be displayed on the button that triggers the click on KInput.
+
+<KCard>
+  <template v-slot:body>
+    <KFileUpload label="Upload File" buttonAppearance="danger" buttonText="Click me" :label-attributes="{ help: 'I use the KLabel `help` prop' }" help="Additional text." icon="close" :accept="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" appearance="file" removable />
+  </template>
+</KCard>
+
+```html
+<KFileUpload label="Upload File" buttonAppearance="danger" buttonText="Click me" :label-attributes="{ help: 'I use the KLabel `help` prop' }" help="Additional text." icon="close" :file-input-accept-types="['.yml', '.yaml', '.json', '.md', '.markdown', 'image/gif', 'image/jpg', 'image/jpeg', 'image/ico', 'image/png']" appearance="file" removable />
+```
+
+### icon
+
+Specify an icon to display to the left of the `placeholder text` if `appearance='image'`.
+
+### iconSize
+
+The size of the `icon` being displayed (default is `26`).
+
+### iconColor
+
+The color of the `icon` being displayed.
+
+### maxFileSize
+
+Use this prop to customize the maximize size of file that can be uploaded. Default value for `image` upload is `1mb (1000000 bytes)` and for `any other file` is `5.24mb (5242880 bytes)`.
```

Comment:
```suggestion
Use this prop to customize the maximize size of file that can be uploaded. Default value for `image` upload is `1mb (1000000 bytes)` and for `file` is `5.24mb (5242880 bytes)`.
```

Can we tell them what happens if they try to upload a file larger than that?

Reply:
it emits 'file-upload-error', do you want me to add an 'example' in 'md' file ?

Reply:
@aanchalm01 instead of `file-upload-error` can we just emit `error` that would be used like this? `<KFileUpload @error="errorCallback" />`